internal/repository/balance.go

```diff
@@ -0,0 +1,164 @@
+package repository
+
+import (
+	"context"
+	"errors"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+
+	"github.com/ivas1ly/gophermart/internal/entity"
+	"github.com/ivas1ly/gophermart/internal/lib/storage/postgres"
+	repoEntity "github.com/ivas1ly/gophermart/internal/repository/entity"
+)
+
+type BalanceRepository struct {
+	db *postgres.DB
+}
+
+func NewBalanceRepository(db *postgres.DB) *BalanceRepository {
+	return &BalanceRepository{
+		db: db,
+	}
+}
+
+func (r *BalanceRepository) GetUserBalance(ctx context.Context, userID string) (*entity.Balance, error) {
+	userBalance := &repoEntity.Balance{}
+
+	query := r.db.Builder.
+		Select("users.id, users.current_balance, COALESCE(SUM(withdrawals.withdrawn), 0)").
+		From("users").
+		LeftJoin("withdrawals ON withdrawals.user_id = users.id").
+		Where(sq.Eq{
+			"users.id": userID,
+		}).
+		GroupBy("users.id")
+
+	sql, args, err := query.ToSql()
+	if err != nil {
+		return nil, err
+	}
+
+	row := r.db.Pool.QueryRow(ctx, sql, args...)
+
+	err = row.Scan(
+		&userBalance.ID,
+		&userBalance.Balance,
+		&userBalance.Withdrawn,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return repoEntity.ToUserBalanceFromRepo(userBalance), nil
+}
+
+func (r *BalanceRepository) NewWithdrawal(ctx context.Context, userID, withdrawalID, orderNumber string,
+	sum int64) error {
```

Comment:
```suggestion
func (r *BalanceRepository) AddWithdrawal(ctx context.Context, wd *entities.Withrowal) error {
```

New означает что дальше будет конструктор
Большое количество параметров тяжело читается